### PR TITLE
code refactor and prepare 1.18.2

### PR DESCRIPTION
**Pull request by wb1016**
refactor like fabric example mod, sodium, campanion.
also cleaned up build.gradle, modrinth automatic upload funtion removed. (i think manual upload is better)
migrated mappings for 1.18.2.
due to unknown imports, help is needed.
```> Configure project :
Fabric Loom: 0.11.32

> Task :compileJava FAILED
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\enchantment\EnchantingBlocks.java:3: error: '.' expected
import F;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\enchantment\EnchantingBlocks.java:4: error: '.' expected
import I;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\enchantment\EnchantmentUtil.java:3: error: '.' expected
import I;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\enchantment\EnchantmentUtil.java:4: error: '.' expected
import Z;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\enchantment\ExperienceUtil.java:3: error: '.' expected
import I;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\enchantment\ExperienceUtil.java:4: error: '.' expected
import J;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\LimitlessLogger.java:3: error: '.' expected
import J;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\transform\mixin\anvil\AnvilScreenMixin.java:3: error: '.' expected
import I;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\transform\mixin\enchantment\EnchantingTableBlockMixin.java:3: error: '.' expected
import I;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\transform\mixin\enchantment\EnchantmentHelperMixin.java:3: error: '.' expected
import Z;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\transform\mixin\enchantment\EnchantmentScreenMixin.java:3: error: '.' expected
import I;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\transform\Transformer.java:13: error: '.' expected
import I;
        ^
C:\Users\wb\minecraft\limitless\src\main\java\net\auoeke\limitless\transform\Transformer.java:14: error: '.' expected
import Z;
        ^
13 errors

FAILURE: Build failed with an exception.
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,65 +1,78 @@
...
-
-    bloated("mod-menu")
+	// To change the versions see the gradle.properties file
+	minecraft "com.mojang:minecraft:${project.minecraft_version}"
+	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
+	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	//fabric api
+	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
+    
+	//modImplementation "net.auoeke:huntingham-hills"
```
Comment (wb1016) [inline build.gradle:33]: why is this needed?

File: build.gradle [hunk 2]
```diff
@@ -1,65 +1,78 @@
...
+    modApi("me.shedaniel.cloth:cloth-config-fabric:${project.cloth_config_version}") {
+		exclude(group: "net.fabricmc.fabric-api")
+	}
+    modImplementation "com.terraformersmc:modmenu:${project.modmenu_version}"
 
-    modCompileOnly("cf:enchantmentDescriptions-250419:3615620")
-    modCompileOnly("me.shedaniel:RoughlyEnoughItems-fabric")
+    //modCompileOnly("cf:enchantmentDescriptions-250419:3615620")
+    //modCompileOnly("me.shedaniel:RoughlyEnoughItems-fabric")
```
Comment (wb1016) [inline build.gradle:40]: same question as above.

File: src/main/java/net/auoeke/limitless/enchantment/EnchantingBlocks.java [hunk 3]
```diff
@@ -1,9 +1,15 @@
 package limitless.enchantment;
 
+import F;
+import I;
```
Comment (wb1016) [inline src/main/java/net/auoeke/limitless/enchantment/EnchantingBlocks.java:4]: what are these one-letter imports?

File: build.gradle (hunk 2 above)
Comment (auoeke) [reply build.gradle:40]: I used them for compatibility testing with Enchantment Descriptions due to a bug.

File: build.gradle (hunk 1 above)
Comment (auoeke) [reply build.gradle:33]: It's a library for transforming classes manually that I use when Mixin breaks or does not do what I need.

File: src/main/java/net/auoeke/limitless/enchantment/EnchantingBlocks.java (hunk 3 above)
Comment (auoeke) [reply src/main/java/net/auoeke/limitless/enchantment/EnchantingBlocks.java:4]: I don't know and they're not in the original source file. I've had Loom produce such garbage while migrating mappings.

Comment (auoeke) [reply src/main/java/net/auoeke/limitless/enchantment/EnchantingBlocks.java:4]: These are primitive type descriptors so I guess that they're derived from some `float` and `int` references in that file.

Comment (wb1016) [reply src/main/java/net/auoeke/limitless/enchantment/EnchantingBlocks.java]: i removed those imports now.

File: build.gradle (hunk 1 above)
Comment (wb1016) [reply build.gradle]: i reverted commenting out, and it is producing errors, how should i resolve the problem?
```A problem occurred configuring root project 'limitless'.
> Could not resolve all dependencies for configuration ':modImplementation'.
   > Could not find net.auoeke:huntingham-hills:.
     Required by:
         project :
```

Comment (auoeke) [reply build.gradle]: It's missing a version. I'll merge it into limitless later so it's not necessary.

**Comment by auoeke** [comment]
> refactor like fabric example mod, sodium, campanion.

Thanks for this but there's much that I dislike about the standard setup. I like the workflow though. I prefer the current directory layout with `source` and `resources` at the root because there's less nesting.

>  ... modrinth automatic upload funtion removed. (i think manual upload is better)

I'll keep the Modrinth upload task.
